### DISCUSSION TITLE
Release for support PHP 8

### DISCUSSION THREAD

**Discussion by hrs-o** [deployphp/deployer#2294] [Help needed]
Any ETA for that?

**Comment by antonmedv** [comment]
I think the current v7 beta already supports php8.

**Reply by hrs-o** [reply]
@antonmedv 

thx!

When I try to add a v7.0.0-beta.10 in the PHP8.0.0RC4 environment, I get an error.

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires deployer/deployer ^v7.0.0-beta.10 -> satisfiable by deployer/deployer[v7.0.0-beta.10].
    - deployer/deployer v7.0.0-beta.10 requires php ^7.3 -> your php version (8.0.0RC4) does not satisfy that requirement.
```

I don't think PHP 8.0 is supported by composer.json.

composer.json

```json
    "require": {
        "php": "^7.3",
        "pimple/pimple": "^3",
        "psr/http-message": "^1",
        "react/http": "^1",
        "symfony/console": "^5",
        "symfony/process": "^5",
        "symfony/yaml": "^5"
    },
```

**Reply by antonmedv** [reply]
Got it. Please verify what at least phar is working.

**Comment by hrs-o** [comment]
It seems that Pimple is not yet compatible with PHP8, and it was difficult to simply update the composer.json.
If you downgrade Pimple to 3.2.3, you may be able to install it by simply updating the composer.json of the deployer.


```sh
    - pimple/pimple is locked to version v3.3.0 and an update of this package was not requested.
    - pimple/pimple v3.3.0 requires php ^7.2.5 -> your php version (8.0.0RC5) does not satisfy that requirement.

---

Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 1 update, 0 removals
  - Downgrading pimple/pimple (v3.3.0 => v3.2.3)

```

https://github.com/silexphp/Pimple/pull/259

**Reply by antonmedv** [reply]
Will work on this too. Will include pimple to Deployer itself.

**Comment by antonmedv** [comment]
Released v7.0.0-beta.11. Should work with PHP8. Please, verify.

**Reply by hrs-o** [reply]
Thank you!!
I have not confirmed the operation, but I have confirmed the installation.

**Comment by garak** [comment]
Any ETA for a stable release?
I see that milestone 7.0 has 0 issues for 25 days: when will be released as stable?

**Reply by antonmedv** [reply]
Currently I’m working docs. As soon as it ready, will release a stable version.
